projects/js-packages/ai-client/src/components/ai-control/index.tsx

```diff
@@ -79,11 +79,26 @@ export function AIControl(
 	ref
 ) {
 	const promptUserInputRef = useRef( null );
+	const InputWrapperRef = useRef( null );
 	const loading = state === 'requesting' || state === 'suggesting';
 
 	// Pass the ref to forwardRef.
 	useImperativeHandle( ref, () => promptUserInputRef.current );
 
+	const focusOnWrapper = () => {
+		// Small delay to avoid focus crash
+		setTimeout( () => {
+			InputWrapperRef.current?.focus?.();
+		}, 100 );
+	};
+
+	const handleSend = useCallback( () => {
+		onSend?.( value );
+		// The input is disabled while the request is executing, so we need
+		// to focus on the wrapper for the stop keyboard shortcut to work
+		focusOnWrapper();
+	}, [ onSend, value ] );
+
 	useKeyboardShortcut(
```

Comment:
The AI Assistant block [already handles the `ESC` keypress event](https://github.com/Automattic/jetpack/blob/8aa42e170d082fdb391a3af1dc667066b7403e9e/projects/plugins/jetpack/extensions/blocks/ai-assistant/edit.js#L395-L405). I think it should be consolidated in the same PR.

Reply:
That event is for the whole block (red part), while the one added at the AIControl component only works on the green part:

![2023-09-25_14-41-24](https://github.com/Automattic/jetpack/assets/8486249/243112c4-cecd-4103-a159-fb76bd161460)

Reply:
so we are binding two functions for the same `ESC` keypress event?